### PR TITLE
Initial Commit

### PR DESCRIPTION
**Pull request by romitkarmakar**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,82 @@
-# hapi-authz
+# Hapi Authz
+
+This is a authorization middleware for Hapi js, and it is based on node-casbin.
+
+## Installation
+
+```shell
+npm i casbin hapi-authz
```
Comment (nodece) [inline README.md]: Use `npm i casbin hapi-authz --save`.

File: README.md [hunk 2]
```diff
@@ -1 +1,82 @@
...
+    return username
+  }
+}
+
+const init = async () => {
+    ...
+    const enforcer = await newEnforcer('model.conf', 'policy.csv') // replace with your model and policy file location
+
+    await server.register({  
+    plugin: require('hapiauthz'),
```
Comment (nodece) [inline README.md]: Should is `require('hapi-authz')`.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    "test": "lab --sourcemaps --transform node_modules/lab-transform-typescript -a @hapi/code"
+  },
+  "keywords": [],
+  "author": "romitkarmakar<romitkarmakar@gmail.com>",
+  "license": "ISC",
+  "dependencies": {
+    "casbin": "^4.1.1"
+  },
+  "devDependencies": {
+    "@hapi/code": "^8.0.1",
```
Comment (nodece) [inline package.json:32]: move to `dependencies`.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    "@hapi/hapi": "^19.1.1",
```
Comment (nodece) [inline package.json]: move to `dependencies`.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    "@hapi/lab": "^22.0.4",
```
Comment (nodece) [inline package.json:33]: move to `dependencies`.

File: src/index.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+      method: ["GET", "POST"],
+      path: "/login",
+      handler: function (request, h) {
+        return "403 Forbidden";
+      }
+    });
+  }
+};
+
+module.exports = hapiauthz;
```
Comment (nodece) [inline src/index.ts]: please use `export`.

File: tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (nodece) [inline tsconfig.json:1]: Please add `declaration ` and `declarationDir`.

File: package.json [hunk 8]
```diff
@@ -0,0 +1,33 @@
+{
```
Comment (nodece) [inline package.json:1]: Please add `typings`, `files`, `repository`.

**Comment by nodece** [comment]
Please refer to https://github.com/casbin/node-casbin to view the TypeScript configuration.

**Comment by romitkarmakar** [comment]
@nodece done all the changes according to your suggestions. Please review it..

File: .travis.yml [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+  - "12"
+  - "node"
+
+sudo: false
+
+install:
+  - "npm install -g codecov"
+  - "npm install"
+  - "npm run tsc"
+
```
Comment (nodece) [inline .travis.yml:14]: Here add `- "npm run test"`.

File: src/BasicAuthorizer.ts [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    return username;
+  }
+
+  // checkPermission checks the user/method/path combination from the request.
+  // Returns true (permission granted) or false (permission forbidden)
+  async checkPermission(): Promise<boolean> {
+    const user = this.getUserName()
+    return await this.enforcer.enforce(user, this.request.path, this.request.method);
+  }
+}
```
Comment (nodece) [inline src/BasicAuthorizer.ts:27]: missing newlines.

File: src/index.ts [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+        return "403 Forbidden";
+      }
+    });
+  }
+};
+
+export {
+  BasicAuthorizer,
+  Hapiauthz
+}
```
Comment (nodece) [inline src/index.ts:42]: missing newlines.

File: package.json (hunk 5 above)
Comment (romitkarmakar) [reply package.json:33]: ```@hapi/lab``` and ```@hapi/code``` is only for running tests. Do I need to migrate it to ```dependencies``` ?

File: src/BasicAuthorizer.ts (hunk 10 above)
Comment (romitkarmakar) [reply src/BasicAuthorizer.ts:27]: Sorry, I didn't understood about newlines. What I have to do here, could you please guide me.

File: tsconfig.json (hunk 7 above)
Comment (romitkarmakar) [reply tsconfig.json:1]: Yes, I did that in the recent commit.

File: package.json (hunk 5 above)
Comment (nodece) [reply package.json:33]: @romitkarmakar You are right.

File: .travis.yml (hunk 9 above)
Comment (romitkarmakar) [reply .travis.yml:14]: Done.

File: src/BasicAuthorizer.ts (hunk 10 above)
Comment (nodece) [reply src/BasicAuthorizer.ts:27]: It means add newlines 28.

**Comment by nodece** [comment]
@romitkarmakar I hope you only keep a commit like: `Init commit`.

**Comment by nodece** [comment]
thanks for the pull request @romitkarmakar. Much appreciated.

**Comment by romitkarmakar** [comment]
Thanks @nodece, for your valuable support.

**Pull request merged**